cockroachdb/cockroach
# release: move email templates to external files

"Week -1" emails are going to have some email content variants depending on various cases, e.g. if it's an alpha, number
of blockers, whether we've passed the prep/qualify date, etc.

This commit sets up pattern for creating email templates for different data variants, and for verifying expected
email output via `.golden` files.

### To generate `.golden` files:
- add template test to `pkg/cmd/release/sender_test.go`
- run `go test -v ./pkg/cmd/release -update`

Actual logic to fetch blockers and wire up data to actual email templates will be implemented in a later commit (PR https://github.com/cockroachdb/cockroach/pull/77250).

### Manual testing

Test 1: confirmed that `post-blockers` sends out email:

```
bazel build //pkg/cmd/release
SMTP_PASSWORD=$MY_SMTP_PWD SMTP_USER=$MY_SMTP_USER GITHUB_TOKEN=$MY_GITHUB_TOKEN \
$(bazel info bazel-bin)/pkg/cmd/release/release_/release post-blockers \
  --release-series 21.2 \
  --smtp-user $MY_SMTP_USER --smtp-host smtp.gmail.com --smtp-port 587 \
  --to $MY_TO_EMAIL \
  --template-dir ./pkg/cmd/release/templates \
  --prep-date 2022-10-11 --publish-date 2022-12-13
```

<img width="550" alt="image" src="https://user-images.githubusercontent.com/3051672/157312141-1f5a98d6-d4a3-47a5-afae-400eb18ff0d3.png">

Test 2: confirmed that `pick-sha` shows me expected s3 errors (see "modifications" section below)


### Modifications to previous release commands:

Note that because emails are now using external templates, `pick-sha` will now require a `template-dir` parameter, i.e.:

```
bazel build //pkg/cmd/release
SMTP_PASSWORD=foo JIRA_USERNAME=foo JIRA_TOKEN=foo \
$(bazel info bazel-bin)/pkg/cmd/release/release_/release pick-sha \
--dry-run --release-series 21.1 \
--smtp-user foo --smtp-host smtp.gmail.com --smtp-port 587 \
--to release@cockroachlabs.com \
--qualify-bucket foo --qualify-object-prefix foo --release-bucket foo --release-object-prefix foo \
--template-dir ./pkg/cmd/release/templates
```


Release note: None
Release justification: Non-production code change.
